### PR TITLE
Adding content view version removal stubs

### PR DESCRIPTION
**Pull request by ldjebran**
```bash
"!!! Congratulations your changes are good to fly, make a great PR! dlezz++ !!!"
```
issue: #1938

### CODE REVIEW

File: tests/foreman/ui/test_contentview.py [hunk 1]
```diff
@@ -3430,3 +3430,367 @@ def test_positive_promote_CV_with_custom_user_role_and_filters(self):
...
+        3. Publish the content view
+
+        4. Promote the content view version from Library -> DEV
+
+        5. remove the content view version from Library environment
+
+        @Assert:
+
+        1. Content view version exist only in DEV and not in Library
+
```
Comment (sghai) [inline tests/foreman/ui/test_contentview.py]: empty lines between steps are not required. and its applicable to other stubs as well.

File: tests/foreman/ui/test_contentview.py [hunk 2]
```diff
@@ -3430,3 +3430,367 @@ def test_positive_promote_CV_with_custom_user_role_and_filters(self):
...
+        3. Publish content view
+
+        4. Promote the content view version to multiple environments
+           Library -> DEV -> QE
+
+        5. remove the content view version from Library environment
+
+        @Assert: Content view version exist only in DEV, QE
+                 and not in Library
```
Comment (sghai) [inline tests/foreman/ui/test_contentview.py]: I think this tab before `and not in Library` is not required. ?

File: tests/foreman/ui/test_contentview.py [hunk 3]
```diff
@@ -3430,3 +3430,367 @@ def test_positive_promote_CV_with_custom_user_role_and_filters(self):
...
+        5. remove the content view version from PROD environment
+
+        @Assert: content view version exists only in Library, DEV, QE, STAGE
+                   and not in PROD
+
+        7. Promote again from STAGE -> PROD
+
+        @Assert: Content view version exist in Library, DEV, QE, STAGE, PROD
+
+
```
Comment (sghai) [inline tests/foreman/ui/test_contentview.py]: one empty line can be removed.

File: tests/foreman/ui/test_contentview.py [hunk 4]
```diff
@@ -3430,3 +3430,367 @@ def test_positive_promote_CV_with_custom_user_role_and_filters(self):
...
+        1. Activation key exists
+
+        2. Content-host exists
+
+        3. Content view does not exist in activation key and content-host
+           association
+
+        @caseautomation: notautomated
+
+        @CaseLevel: Integration
```
Comment (sghai) [inline tests/foreman/ui/test_contentview.py]: since tier3 so `@CaseLevel` should be `System`

File: tests/foreman/ui/test_contentview.py [hunk 5]
```diff
@@ -3430,3 +3430,367 @@ def test_positive_promote_CV_with_custom_user_role_and_filters(self):
...
+        5. Promote the content view to multiple environment
+           Library -> DEV -> QE -> PROD
+
+        6. Disconnect the capsule
+
+        7. Remove the content view version from Library and DEV environments
+           and assert successful completion
+
+        8. Bring the capsule back online and assert that the task is completed
+           in capsule
```
Comment (sghai) [inline tests/foreman/ui/test_contentview.py]: Interesting scenario.

**Comment by sghai** [comment]
@ldjebran : have you tried all these scenarios manually ? Just asking because even I'm not sure how server would behave in some cases, like the last stub

**Comment by ldjebran** [comment]
@sghai , No, the majority of tests seems logical and tried them in daily work with sat.
delete CV with subscribed host not tested, let me test it manually and return back. 
But latest I even have to investigate on how to make the necessary setups.
**let me make this PR in progress back in order to make the necessary investigations**

**Comment by ldjebran** [comment]
@sghai , content-host stub tests steps need update, perhaps the issue was written in an earlier stage

File: tests/foreman/ui/test_contentview.py [hunk 6]
```diff
@@ -3675,36 +3640,32 @@ def test_positive_remove_cv_version_from_env_with_host_registered(self):
...
-        7. Remove the content view version from QE environment
-
-        8. Verify if the affected Activation key and content-host are
-           functioning properly
+        7. Remove the content view cv1 version from QE environment.
+           The remove environment wizard should propose to replace the current
+           QE environment of cv1 by an other (as QE environment of cv1
+           is attached to a content-host),
+           choose DEV and content view cv1 as a replacement for Content-host
+           and for Activation key.
```
Comment (sghai) [inline tests/foreman/ui/test_contentview.py]: As per discussion, need to add step to refresh the subscription from content-hosts..

File: tests/foreman/ui/test_contentview.py [hunk 7]
```diff
@@ -3675,36 +3640,32 @@ def test_positive_remove_cv_version_from_env_with_host_registered(self):
...
         1. Activation key exists
-
         2. Content-host exists
-
-        3. Content view does not exist in activation key and content-host
-           association
+        3. QE environment of cv1 was replaced by DEV environment of cv1
+           in activation key
+        4. QE environment of cv1 was replaced by DEV environment of cv1
+           in content-host
```
Comment (sghai) [inline tests/foreman/ui/test_contentview.py]: and here need to again fetch an rpm via yum

File: tests/foreman/ui/test_contentview.py [hunk 8]
```diff
@@ -3766,31 +3720,23 @@ def test_positive_remove_cv_version_from_multi_env_capsule_scenario(self):
...
         9. Make sure the capsule is updated
 
         @Assert: content view version in capsule is removed from Library
-                 and DEV and exists only in QE and PROD
+        and DEV and exists only in QE and PROD
 
         @caseautomation: notautomated
 
-        @CaseLevel: Integration
+        @CaseLevel: System
```
Comment (sghai) [inline tests/foreman/ui/test_contentview.py]: This test case requires complete external capsule configuration.

Comment (ldjebran) [inline tests/foreman/ui/test_contentview.py]: @sghai,  so what's CaseLevel in this situation?

**Comment by ldjebran** [comment]
@sghai, as discussed, updated and mare PR back  to review

File: tests/foreman/ui/test_contentview.py [hunk 9]
```diff
@@ -3766,31 +3720,23 @@ def test_positive_remove_cv_version_from_multi_env_capsule_scenario(self):
         @Steps:
 
         1. Create a content view
-
         2. Setup satellite to use a capsule and to sync all lifecycle
```
Comment (sghai) [inline tests/foreman/ui/test_contentview.py]: Usually, its not a common practice to sync all environments to capsule and especially the `Library` as it includes all contents. I think we need to revisit the test once we will have implementation around capsule in place.

**Comment by sghai** [comment]
ACK.. 

thank you @ldjebran for adding these stubs. I like all of them specially they way we are covering all kind of contents in CV. 

However, I still feel that the last stub related to capsule needs some refinement but its not a blocker. We can revisit when we start implementing it.

**Comment by ldjebran** [comment]
@sghai , many thanks, I have manually tested the latest test and it seems that this a reliably working scenario, but with a reliable way to create a capsule host, that is out of the test area, but we should perhaps create some base function that can do it using the existing automation libs/tools in a new issue

**Pull request merged**